utils/unittest/epoch_builder.go

```diff
@@ -94,9 +94,11 @@ func (builder *EpochBuilder) BuildEpoch() *EpochBuilder {
 	B.SetPayload(flow.Payload{Seals: seals})
 	err = builder.state.Mutate().Extend(&B)
 	require.Nil(builder.t, err)
-	// finalize block B
+	// finalize and validate block B
 	err = builder.state.Mutate().Finalize(B.ID())
 	require.Nil(builder.t, err)
+	err = builder.state.Mutate().MarkValid(B.ID())
+	require.Nil(builder.t, err)
```

Comment:
you may use `require.NoError(builder.t, err)`. It gets more indicative logs on failure, especially over CI.

Reply:
Fair point. I've been using `Nil` because it's functionally the same and symmetric with how we check errors in non-test code, but you're right `NoError` will print out the error string which is a nice help for debugging 👍